src/app/projects/project-list-filters/project-list-filters.component.ts

```diff
@@ -0,0 +1,153 @@
+  @Input() projects: Array<Project> = []; // from project list component
+  @Input() tableParams: TableParamsObject = new TableParamsObject(); // from project list component
+  @Output() updateMatching = new EventEmitter<UpdateMatchingData>(); // to project list component
+
+  public loading: boolean = false;
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+  public applyFilterCount: number = 0;
+
+	public projectTypeFilters: ProjectType[] = [
+		{name: 'Land Use Planning', checked: true},
+		{name: 'Forest Landscape Planning', checked: true},
+		{name: 'Water Planning and Governance', checked: true}
+	];
+
+  public fullNames: ProjectTypeFullNameMap = {
+    lup: 'Land Use Planning',
+    flp: 'Forest Landscape Planning',
+    wpag: 'Water Planning and Governance',
+  }
+
+  constructor(
+    public commentPeriodService: CommentPeriodService, // also used in template
+  ) {}
+
+  // Apply all filters and emit data to parent component
+  public ngOnInit() {
+    this.internalApplyAllFilters();
+  }
+
+  // Triggered when the change detection is triggered from the parent component.
+  public ngOnChanges() {
+    this.internalApplyAllFilters('search');
+  }
+
+  /**
+   * Applies the current this.tableParams values to the filter controls.
+   */
+  public applyParamsToFilterControls() {
+    if (this.tableParams.projectTypes) {
+      const projectTypesArray = this.tableParams.projectTypes.split(',');
+      const expandedProjectTypesArray = projectTypesArray.map(item => this.fullNames[item] || item);
+      this.projectTypeFilters.forEach(ptf => {
+        ptf.checked = expandedProjectTypesArray.includes(ptf.name) || 'all' === this.tableParams.projectTypes ? true : false;
+      })
+    }
+    this.applyFilterCount++;
+  }
+
+	/**
+	 * Handles an event change to the project type filter
+	 * 
+	 * @param {MatCheckboxChange} event The event that is passed from the project type filter checkbox
+	 * @returns {void}
+	 */
+	public handleProjectTypeChange(event: MatCheckboxChange): void {
+		const index = this.projectTypeFilters.map(filter => filter.name).indexOf(event.source.value);
+		if (-1 !== index) {
+			this.projectTypeFilters[index].checked = event.checked;
+			this.internalApplyAllFilters('project type filter');
+		}
+	}
+
+  /**
+   * Applies filters.
+   * 
+   * @param source The source of the call to this method (optional)
+	 * @returns {void}
+   */
+  private internalApplyAllFilters(source: string = 'filter'): void {
+    // This runs before ngOnInit, so we need set the filter controls on the first iteration
+    if (this.applyFilterCount === 0) {
+      this.applyParamsToFilterControls();
+    }
+    this.projects.forEach(proj => proj.isMatches = this.showThisProject(proj));
+
+    // Define an array of abreviated project types, based on which ones are active
+    const activeFilters = this.projectTypeFilters
+      .filter(ptf => ptf.checked)
+      .map(ptf => ptf.name.match(/\b\w/g).join('').toLowerCase());
```

Comment:
Wouldn't this be a good place to use `this.fullNames` ?

Reply:
It's a fair point, I could use the value to find the key. I figure it's about equal in processing power to grabbing the first letter of each word and shifting to lowercase. Of course, when going from abbreviated form to long form, the name map becomes essential.